ci/test.sh

```diff
@@ -118,3 +118,77 @@ assert_no_diff
 
 # Run the integration tests
 (cd bindgen-integration && cargo test $CARGO_ARGS)
+
+if [ "$BINDGEN_RUST_FOR_LINUX_TEST" == "1" ]; then
+  # Run the Rust for Linux test
+  #
+  # This is intended to be an end-to-end test that runs what Linux kernel
+  # developers/users would do. It is a quick, build-only test of the Rust code
+  # in the Linux kernel.
+
+  # Put LLVM binaries in the path for `LLVM=1`. The LLVM `bin` directory should
+  # go first since there are others in the Ubuntu image.
+  export PATH="${LLVM_DIRECTORY}/bin:${PATH}"
+
+  # Kernel build dependency: `bindgen-cli`, which is under test.
+  #
+  # Using the unstable `--out-dir` so that we can use `cargo build` instead of `cargo install`,
+  # since we want to reuse the `$CARGO_ARGS`.
+  (cd bindgen-cli && RUSTC_BOOTSTRAP=1 cargo build -Zunstable-options --out-dir ${HOME}/.bindgen $CARGO_ARGS)
+  export PATH="${HOME}/.bindgen:${PATH}"
+
+  # Kernel build dependency: `libelf-dev`.
+  sudo apt-get update
+  sudo apt-get install libelf-dev
+
+  # Kernel build dependency: the Rust standard library sources.
+  #
+  # `rustup` is used here to install the `rust-src` component (instead of using
+  # `actions-rs/toolchain`'s `components` option in the workflow step) since we
+  # only need it for this test, and anyway the action installs `rustup`.
+  rustup component add rust-src
+
+  # Ideally this should be updated from time to time (e.g. every kernel `-rc1`),
+  # and each update should only contain this change.
+  #
+  # Both commit hashes and tags are supported.
+  LINUX_VERSION=c13320499ba0efd93174ef6462ae8a7a2933f6e7
+
+  # Download Linux at a specific commit
+  mkdir -p linux
+  git -C linux init
+  git -C linux remote add origin https://github.com/torvalds/linux.git
+  git -C linux fetch --depth 1 origin ${LINUX_VERSION}
+  git -C linux checkout FETCH_HEAD
+
+  # Configure Rust for Linux
+  cat <<EOF > linux/kernel/configs/rfl-for-bindgen-ci.config
+# CONFIG_WERROR is not set
```

Comment:
This could be reduced in scope a bit more, but I just did what we are currently doing in the Rust CI, which seemed reasonable (and they should notice any issue there first).